runtime/shibuya/src/lib.rs

```diff
@@ -248,7 +249,7 @@ parameter_types! {
     pub RuntimeBlockWeights: BlockWeights = BlockWeights::builder()
         .base_block(BlockExecutionWeight::get())
         .for_class(DispatchClass::all(), |weights| {
-            weights.base_extrinsic = ExtrinsicBaseWeight::get();
+            weights.base_extrinsic = weights::base_extrinsic::ExtrinsicBaseWeight::get();
```

Comment:
I don't think we need this. Why not use ExtrinsicBaseWeight and add 2 extra reads. We don't have to maintain this benchmark. Also make sure the fee calculation reflects this

Reply:
I recommended to use this so I can answer - we benchmark everything else essentially, relying on automated measurements when possible.

It's easier to have something like this, then to hardcode it, IMO.
More correct & consistent also.

Shouldn't be that much of a maintenance since only an additional command is executed within an existing loop, right?

Reply:
@Dinonard if that's the case then benchmark is missing reads

Reply:
@ipapandinas @Dinonard this overhead benchmark doesn't count any IO. It just measures the overhead of an extrinsic but not the actual execution of the extrinsic. So there's no use for this. We just need to manually add 2 reads and it should be fine

Reply:
I've checked the overhead benchmark command code.

Remark extrinsics pushed into the block used during the benchmark are executed in the client runtime via the `FrameExecutive::apply_extrinsic`. These extrinsics are dispatched, and the `FrameSystem::BaseCallFilter` is checked alongside the origin. The new `SafeMode` & `TxPause` filters are then measured.

When running the command locally, here are the results I obtained:

#### Without base call filtering:

`type BaseCallFilter = frame_support::traits::Everything;`
Average Result:  **95_705** nanoseconds

#### With base call filtering:

`type BaseCallFilter = InsideBoth<SafeMode, TxPause>;`
Average Result:  **103_702** nanoseconds

This represents an 8% increase.
PoV is also measured - reference: https://github.com/paritytech/substrate/pull/11637

Reply:
However I have the following error when performing the overhead benchmark command via `shibuya-dav` in the Github workflow:

```bash
Finished release [optimized] target(s) in 38m 52s
2024-12-02 14:51:56 Using the chain spec instead of the runtime to generate the genesis state is deprecated. Please remove the `--chain`/`--dev`/`--local` argument, point `--runtime` to your runtime blob and set `--genesis-builder=runtime`. This warning may become a hard error any time after December 2024.    
2024-12-02 14:51:57 assembling new collators for new session 0 at #0    
2024-12-02 14:51:57 assembling new collators for new session 1 at #0    
2024-12-02 14:51:57 Loading WASM from genesis state    
[+] Benchmarking 1 Astar collator pallets.
[+] Benchmarking pallet_tx_pause with weight file ./benchmark-results/shibuya-dev/tx_pause_weights.rs
[+] Benchmarking block and extrinsic overheads...
2024-12-02 14:52:03 assembling new collators for new session 0 at #0    
2024-12-02 14:52:03 assembling new collators for new session 1 at #0    
2024-12-02 14:52:05 🔨 Initializing Genesis block/state (state: 0x81fd…3aae, header-hash: 0x69b1…ac53)    
2024-12-02 14:52:05 panicked at /home/astar-admin/.cargo/git/checkouts/polkadot-sdk-cff69157b985ed76/b98e0b3/cumulus/pallets/parachain-system/src/lib.rs:1297:30:
included head not present in relay storage proof    
Error: Client(RuntimeApiError(Application(Execution(AbortedDueToTrap(MessageWithBacktrace { message: "wasm trap: wasm `unreachable` instruction executed", backtrace: Some(Backtrace { backtrace_string: "error while executing at wasm backtrace:\n    0: 0x1680a - <unknown>!rust_begin_unwind\n    1: 0x52f5 - <unknown>!core::panicking::panic_fmt::hc677f413e2f2c888\n    2: 0x6c74b0 - <unknown>!frame_support::storage::transactional::with_transaction::hdd4b19daa5635076\n    3: 0x19b9ee - <unknown>!<cumulus_pallet_parachain_system::pallet::Call<T> as frame_support::traits::dispatch::UnfilteredDispatchable>::dispatch_bypass_filter::{{closure}}::hf5e7b72922057598\n    4: 0x15bf8b - <unknown>!<shibuya_runtime::RuntimeCall as frame_support::traits::dispatch::UnfilteredDispatchable>::dispatch_bypass_filter::hb7c74cdeb42c4a6a\n    5: 0x154f5b - <unknown>!<shibuya_runtime::RuntimeCall as sp_runtime::traits::Dispatchable>::dispatch::he9dab724d3fc532c\n    6: 0x5d1038 - <unknown>!frame_executive::Executive<System,Block,Context,UnsignedValidator,AllPalletsWithSystem,COnRuntimeUpgrade>::apply_extrinsic::hfe4db20b64a1ce4c\n    7: 0x61f955 - <unknown>!BlockBuilder_apply_extrinsic" }) })))))
2024-12-02 14:52:05 1 storage transactions are left open by the runtime. Those will be rolled back.    
2024-12-02 14:52:05 1 storage transactions are left open by the runtime. Those will be rolled back.    
[-] Failed to benchmark the block and extrinsic overheads. Error written to ./benchmark-results/shibuya-dev/bench_errors.txt; continuing...
[+] Benchmarking the machine...
[-] Benchmarks failed:  ./benchmark-results/shibuya-dev/bench_errors.txt
```